internal/lm/nvml.go

```diff
@@ -193,3 +201,59 @@ func isMPSCapable(manager resource.Manager) (bool, error) {
 	}
 	return true, nil
 }
+
+func newGPUModeLabeler(manager resource.Manager) (Labeler, error) {
+	devices, err := manager.GetDevices()
+	if err != nil {
+		return nil, err
+	}
+	if len(devices) == 0 {
+		// no devices, return empty labels
+		return empty{}, nil
+	}
+	classes, err := getDeviceClasses(devices)
+	if err != nil {
+		return nil, err
+	}
+	gpuMode := getModeForClasses(classes)
+	labels := Labels{
+		"nvidia.com/gpu.mode": gpuMode,
```

Comment:
Question: Since we also extract this label for a MIG device do we expect different labels per MIG profile?

Reply:
Devices that support MIG do not support changing the GPU mode. They are always going to return compute PCI class. We can actually hard code the class of MIG devices to be compute.

Reply:
Yes, let's rather do that. It would simplify the implementation quite a bit.

Reply:
Updated the implementation